### PR TITLE
fix infinite loop

### PR DESCRIPTION
**Pull request by rbren**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

the `/authenticate` endpoint can trigger an infinite loop 😬 

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:b6dd1da-nikolaik   --name openhands-app-b6dd1da   docker.all-hands.dev/all-hands-ai/openhands:b6dd1da
```

### CODE REVIEW

**Review by tofarr — APPROVED** [review]
🤦 - Good catch

File: frontend/__tests__/utils/organizeModelsAndProviders.test.ts [hunk 1]
```diff
@@ -15,7 +15,6 @@ test("organizeModelsAndProviders", () => {
     "gpt-4o",
     "together-ai-21.1b-41b",
     "gpt-4o-mini",
-    "claude-3-5-sonnet-20241022",
```
Comment (xingyaoww) [inline frontend/__tests__/utils/organizeModelsAndProviders.test.ts:18]: Is this intended?

File: frontend/__tests__/utils/organizeModelsAndProviders.test.ts [hunk 2]
```diff
@@ -51,7 +50,6 @@ test("organizeModelsAndProviders", () => {
     anthropic: {
       separator: "/",
       models: [
-        "claude-3-5-sonnet-20241022",
```
Comment (xingyaoww) [inline frontend/__tests__/utils/organizeModelsAndProviders.test.ts:54]: ditto

**Review by amanape — APPROVED** [review]
LGTM

**Pull request merged**